core/src/main/java/org/springframework/security/oauth2/server/authorization/token/TokenIssuer.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.server.authorization.token;
+
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+
+/**
+ * @author Alexey Nesterov
+ */
+public interface TokenIssuer<PRINCIPAL> {
```

Comment:
Please remove this as it's not needed as part of this issue. The `OAuth2ClientCredentialsAuthenticationProvider` should simply leverage `StringKeyGenerator accessTokenGenerator` the same as in `OAuth2AuthorizationCodeAuthenticationProvider`